dm/worker/source_worker_test.go

```diff
@@ -418,12 +422,14 @@ func (t *testWorkerEtcdCompact) SetUpSuite(c *C) {
 		mockSync := NewMockUnit(pb.UnitType_Sync)
 		return []unit.Unit{mockDumper, mockLoader, mockSync}
 	}
+	c.Assert(failpoint.Enable("github.com/pingcap/dm/dm/worker/MockGetSourceCfgFromETCD", `return(true)`), IsNil)
```

Comment:
why we need so many mock?

Reply:
because no data is actually written to ectd in these unit tests, the latest source cfg from etcd is not available